vertical-pod-autoscaler/e2e/v1/full_vpa.go

```diff
@@ -135,18 +179,19 @@ func waitForPodsMatch(f *framework.Framework, listOptions metav1.ListOptions, ma
 			return false, nil
 		}
 
+		result := true
 		for _, pod := range podList.Items {
 			if !matcher(pod) {
-				return false, nil
+				result = false
```

Comment:
Why the return command was removed?

Reply:
I moved it to after the loop. This is to evaluate all the pods (instead of exiting after the first pod that doesn't match). It makes debugging easier (for example test was failing because it didn't filter out failed pods. Which was hard to notice when we logged only about the first pod on the list)

Reply:
Please add comment why we want to have it this way.
TBH, it looks weird without a backstory.

Reply:
Done